cpu/stm32f1/periph/spi.c

```diff
@@ -210,27 +211,35 @@ int spi_transfer_byte(spi_t dev, char out, char *in)
             return -1;
     }
 
-    while (!(spi->SR & SPI_SR_TXE)) {}
-    spi->DR = out;
-    transferred++;
-
-    while (!(spi->SR & SPI_SR_RXNE)) {}
-    if (in != NULL) {
-        *in = spi->DR;
-        transferred++;
-    }
-    else {
+    if(!in){
+        for (unsigned i = 0; i < length; i++) {
+            while (!(spi->SR & SPI_SR_TXE)) {}
+            spi->DR = (uint8_t)out[i];
+            transferred++;
+        }
+        /* SPI busy */
+        while ((spi->SR & SPI_SR_BSY)) {}
         spi->DR;
     }
+    else{
+        char tmp;
+        for (unsigned i = 0; i < length; i++) {
+            tmp = (out) ? out[i] : 0;
+            while (!(spi->SR & SPI_SR_TXE)) {}
+            spi->DR = tmp;
+            transferred++;
+            while (!(spi->SR & SPI_SR_RXNE)) {}
+            in[i] = (char)spi->DR;
+            transferred++;
+        }
+    }
 
-    /* SPI busy */
-    while ((spi->SR & 0x80)) {}
 #if ENABLE_DEBUG
     if (in != NULL) {
-        DEBUG("\nout: %x in: %x transferred: %x\n", out, *in, transferred);
+        DEBUG("\nSPI: transferred %i Bytes\n", transferred);
     }
     else {
-        DEBUG("\nout: %x in: was nullPointer transferred: %x\n", out, transferred);
+        DEBUG("\nSPI: 'in' NullPointer. Transferred %i Bytes\n", transferred);
     }
 #endif /*ENABLE_DEBUG */
```

Comment:
Maybe you can merge those two now that you only have a parameter to print and remove the `if`